docs/tutorial/flask.rst

```diff
@@ -0,0 +1,535 @@
+
+.. code-block::
+
+    Model              Controller      Cloud/Region        Version  SLA          Timestamp
+    flask-hello-world  dev-controller  microk8s/localhost  3.1.8    unsupported  17:04:11+10:00
+
+    App           Version  Status  Scale  Charm              Channel  Rev  Address         Exposed  Message
+    flask-hello-world      active      1  flask-hello-world             0  10.152.183.166  no
+
+    Unit             Workload  Agent  Address      Ports  Message
+    flask-hello-world/0*  active    idle   10.1.87.213
+
+The deployment is finished when the status shows ``active``. Let's expose the
+application using ingress. Deploy the ``nginx-ingress-integrator`` charm and integrate
+it with the Flask app:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:deploy-nginx]
+    :end-before: [docs:deploy-nginx-end]
+    :dedent: 2
+
+The hostname of the app needs to be defined so that it is accessible via the ingress.
+We will also set the default route to be the root endpoint:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:config-nginx]
+    :end-before: [docs:config-nginx-end]
+    :dedent: 2
+
+Monitor ``juju status`` until everything has a status of ``active``. Test the
+deployment using
+``curl http://flask-hello-world --resolve flask-hello-world:80:127.0.0.1`` to send
+a request via the ingress to the root endpoint. It should still be returning
+the ``Hello, world!`` greeting.
+
+.. note::
+
+    The ``--resolve flask-hello-world:80:127.0.0.1`` option to the ``curl``
+    command is a way of resolving the hostname of the request without
+    setting a DNS record.
+
+Configure the Flask application
+===============================
+
+Now let's customise the greeting using a configuration option. We will expect this
+configuration option to be available in the Flask app configuration under the
+keyword ``GREETING``. Go back out to the root directory of the project using
+``cd ..`` and copy the following code into ``app.py``:
+
+.. literalinclude:: code/flask/greeting_app.py
+    :language: python
+
+Open ``rockcraft.yaml`` and update the version to ``0.2``. Run ``rockcraft pack``
+again, then upload the new OCI image to the MicroK8s registry:
+
+.. literalinclude:: code/flask/task.yaml
+    :language: bash
+    :start-after: [docs:docker-update]
+    :end-before: [docs:docker-update-end]
+    :dedent: 2
+
+Change back into the charm directory using ``cd charm``. The ``flask-framework``
+Charmcraft extension supports adding configurations to ``charmcraft.yaml`` which
+will be passed as environment variables to the Flask application. Add the
+following to the end of the ``charmcraft.yaml`` file:
+
+.. code-block:: yaml
+
+    config:
+      options:
+        greeting:
+          description: |
+            The greeting to be returned by the Flask application.
+          default: "Hello, world!"
+          type: string
```

Comment:
For some reason the rendering of this is not quite right
![Screenshot 2024-11-25 at 1 22 00 pm](https://github.com/user-attachments/assets/c87e4911-a32e-4ec8-bc47-f0f39306d572)

Reply:
Yes, it seems like adding more spaces that seemingly necessary fixed the rendering issue...very strange. Thanks again for catching!

Reply:
Ok looks good now, thanks!